### PR TITLE
Clean up `ControlThrowable`.

### PR DESCRIPTION
**Pull request by sjrd**
In bd368a7e336b99d299681c96415fde75b025a92a, `ControlThrowable` received a legacy deprecated constructor to allow enabling writable stack traces, because it was deemed necessary not to break Scala.js. However, Scala.js found a more fundamental way to keep functioning, without this legacy constructor, in https://github.com/scala-js/scala-js/commit/03faf11c670ed0c8eb973e792a771a91c265c320

Therefore, this commit removes that legacy constructor.

Basically cleaning up my own mess ;)

### CODE REVIEW

**Review by som-snytt — APPROVED** [review]
Looks OK to me. It's difficult to keep up with the latest research on exceptions. I suspect that our tormentor in the first circle of the pit will taunt us with this class.

**Review by SethTisue — APPROVED** [review]
LGTM, final review/merge by @NthPortal ?

**Review by NthPortal — APPROVED** [review]
Nice!
